src/dll.py

```diff
@@ -63,6 +65,8 @@ def pop(self):
         if self.head_node:
             self.head_node.previous_node = None
         self.length_list -= 1
+        if len(self) == 0:
```

Comment:
All your code is working; however ideally you would expose and handle this case with a try/except block, rather than relying on asking your `self.length_list` value. Your `self.length_list` is a liability because you have to diligently update it everywhere else in the code.